### ISSUE TITLE
Enum Names Not Overriden with @JsonProperty

### ISSUE THREAD

**Issue by PeaceInColour** [swagger-api/swagger-core#2707]
| Q                                 | A
| --------------------------------- | -------
| Bug or feature request?           | Bug
| Which Swagger-Core version?       |1.5.18
| Which Java version?               |1.8u151
| Which JAX-RS framework & version? |RESTEasy 3.0.19 with JAX-RS 2.0
 

Using `@JsonProperty` to override enum names isn't working. The following example below:

```java
public enum SomeEnum {
	@JsonProperty("num-one")
	NUMONE,
	
	@JsonProperty("num-two")
	NUMTWO
}
```
results in an OpenAPI file that looks like:

```json
"enum": [
	"NUMONE",
	"NUMTWO"
]
```

instead of the expected:

```json
"enum": [
	"num-one",
	"num-two"
]
```

The issue is in ModelResolver which is using the deprecated Jackson Databind method `findEnumValue` to read in the enum name. This method does not look for any overrides and just defaults to the name in the code. Using the replacement method `findEnumValues` fixes this problem.

**Comment by PeaceInColour** [comment]
Once I get permission to push (or figure out what's wrong if the issue is on my end) I can push a fix for this.

**Comment by frantuma** [comment]
Thanks for the fix! to open a PR, please fork the repo and push the branch to your own repo, and open the PR from there targeting the main repo.

**Comment by frantuma** [comment]
fixed in #2796

**Comment by rabiori** [comment]
any workarounds for someone stuck in the old version

**Comment by jlous** [comment]
@rabiori Ever find a workaround?

**Comment by DeepakPatankar** [comment]
I am also stuck in the old version, can anyone suggest a workaround?

**Comment by jlous** [comment]
We used a custom ModelConverter on 2.1.0, which has a regression on this issue (fixed in 2.1.3)
Don't know if that option applies to 1.5
